sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/filters/TimeWindowFilter.java

```diff
@@ -35,38 +43,36 @@ public TimeWindowFilter() {
      */
     @Override
     public boolean evaluate(FeatureFilterEvaluationContext context) {
-        String start = (String) context.getParameters().get(TIME_WINDOW_FILTER_SETTING_START);
-        String end = (String) context.getParameters().get(TIME_WINDOW_FILTER_SETTING_END);
+        final Map<String, Object> parameters = context.getParameters();
+        final Object recurrenceObject = parameters.get(TIME_WINDOW_FILTER_SETTING_RECURRENCE);
+        if (recurrenceObject != null) {
+            final Map<String, Object> recurrenceParameters = (Map<String, Object>) recurrenceObject;
+            final Object patternObj = recurrenceParameters.get(RecurrenceConstants.RECURRENCE_PATTERN);
+            if (patternObj != null) {
+                FeatureFilterUtils.updateValueFromMapToList((Map<String, Object>) patternObj, RecurrenceConstants.RECURRENCE_PATTERN_DAYS_OF_WEEK);
+            }
+        }
 
-        ZonedDateTime now = ZonedDateTime.now();
+        final TimeWindowFilterSettings settings = OBJECT_MAPPER.convertValue(context.getParameters(), TimeWindowFilterSettings.class);
+        final ZonedDateTime now = ZonedDateTime.now();
 
-        if (!StringUtils.hasText(start) && !StringUtils.hasText(end)) {
+        if (settings.getStart() == null && settings.getEnd() == null) {
             LOGGER.warn("The {} feature filter is not valid for feature {}. It must specify either {}, {}, or both.",
                 this.getClass().getSimpleName(), context.getName(), TIME_WINDOW_FILTER_SETTING_START,
                 TIME_WINDOW_FILTER_SETTING_END);
             return false;
         }
-        
-        ZonedDateTime startTime = null;
-        ZonedDateTime endTime = null;
 
-        try {
-            startTime = StringUtils.hasText(start)
-                ? ZonedDateTime.parse(start, DateTimeFormatter.ISO_DATE_TIME)
-                : null;
-            endTime = StringUtils.hasText(end)
-                ? ZonedDateTime.parse(end, DateTimeFormatter.ISO_DATE_TIME)
-                : null;
-        } catch (DateTimeParseException e) {
-            startTime = StringUtils.hasText(start)
-                ? ZonedDateTime.parse(start, DateTimeFormatter.RFC_1123_DATE_TIME)
-                : null;
-            endTime = StringUtils.hasText(end)
-                ? ZonedDateTime.parse(end, DateTimeFormatter.RFC_1123_DATE_TIME)
-                : null;
+        if (settings.getStart() == null) {
+            return now.isBefore(settings.getEnd());
+        }
+        if (settings.getEnd() == null) {
+            return now.isAfter(settings.getStart());
+        }
+        if (settings.getRecurrence() == null) {
```

Comment:
If one of `Start` and `End` is missing but `Recurrence` is not null, this settings should be considered as invalid. Did you handle it anywhere?

Reply:
No. For this case, we will return `now.isAfter(settings.getStart())` || `now.isBefore(settings.getEnd())`.
That sounds reasonable that we mark this case as invalid. Will add a check for this